packages/rln/src/rln.ts

```diff
@@ -16,266 +15,59 @@ import {
-    options: StartRLNOptions,
-    credentials: KeystoreEntity | undefined
-  ): Promise<StartRLNOptions> {
-    let chainId = credentials?.membership.chainId;
-    const address =
-      credentials?.membership.address ||
-      options.address ||
-      LINEA_CONTRACT.address;
-
-    if (address === LINEA_CONTRACT.address) {
-      chainId = LINEA_CONTRACT.chainId;
-    }
+      const stringEncoder = new TextEncoder();
+      const vkey = stringEncoder.encode(JSON.stringify(verificationKey));
 
-    const signer = options.signer || (await extractMetaMaskSigner());
-    const currentChainId = await signer.getChainId();
+      const DEPTH = 20;
+      const zkRLN = zerokitRLN.newRLN(DEPTH, zkey, vkey);
+      const zerokit = new Zerokit(zkRLN, witnessCalculator, DEFAULT_RATE_LIMIT);
 
-    if (chainId && chainId !== currentChainId) {
-      throw Error(
-        `Failed to start RLN contract, chain ID of contract is different from current one: contract-${chainId}, current network-${currentChainId}`
-      );
+      return new RLNInstance(zerokit);
+    } catch (error) {
+      log.error("Failed to initialize RLN:", error);
+      throw error;
     }
-
-    return {
-      signer,
-      address
-    };
   }
-
-  private static async decryptCredentialsIfNeeded(
-    credentials?: EncryptedCredentials | DecryptedCredentials
-  ): Promise<{ credentials?: DecryptedCredentials; keystore?: Keystore }> {
-    if (!credentials) {
-      return {};
-    }
-
-    if ("identity" in credentials) {
-      return { credentials };
-    }
-
-    const keystore = Keystore.fromString(credentials.keystore);
-
-    if (!keystore) {
-      return {};
-    }
-
-    const decryptedCredentials = await keystore.readCredential(
-      credentials.id,
-      credentials.password
-    );
-
-    return {
-      keystore,
-      credentials: decryptedCredentials
-    };
-  }
-
-  public async registerMembership(
-    options: RegisterMembershipOptions
-  ): Promise<undefined | DecryptedCredentials> {
-    if (!this.contract) {
-      throw Error("RLN Contract is not initialized.");
-    }
-
-    let identity = "identity" in options && options.identity;
-
-    if ("signature" in options) {
-      identity = this.zerokit.generateSeededIdentityCredential(
-        options.signature
-      );
-    }
-
-    if (!identity) {
-      throw Error("Missing signature or identity to register membership.");
-    }
-
-    return this.contract.registerWithIdentity(identity);
-  }
-
-  /**
-   * Changes credentials in use by relying on provided Keystore earlier in rln.start
-   * @param id: string, hash of credentials to select from Keystore
-   * @param password: string or bytes to use to decrypt credentials from Keystore
-   */
-  public async useCredentials(id: string, password: Password): Promise<void> {
-    this._credentials = await this.keystore?.readCredential(id, password);
+  private constructor(public zerokit: Zerokit) {
+    super(zerokit);
```

Comment:
We discussed `composition` vs `extension` for classes. 
https://github.com/waku-org/js-waku/issues/2033

I propose to use primarily `composition` as this is usually cleaner. Exception can be `EvenTarget` but even that I would make a separate property following what `js-libp2p` does:
https://github.com/libp2p/js-libp2p/blob/f04e18253ad9fd6ddcfa5c5fb3afc924408459e4/packages/libp2p/src/libp2p.ts#L98

No immediate action, adding it as a point for discussion for next PM.

For this case we can do a follow up PR.

fyi @waku-org/js-waku

Reply:
here it makes sense to `extend`, simply because we get access to the methods through the root object, instead of accessing sub-object 

let's continue discussions on this!

Reply:
added a discussion https://discord.com/channels/1110799176264056863/1360270396158509106/1360270399069225167